Updated performance workflow docs. (#383)
CitrineInformatics/citrine-python: docs/source/workflows/performance_workflows.rst

=====================

A :class:`performance workflow <citrine.informatics.workflows.PerformanceWorkflow>` performs analysis on a module.
Each analysis computes one or more performance metrics, e.g. accuracy of an ML predictor.
An analysis is codified by a configuration object that stores all relevant settings and parameters required to run the workflow.
By storing the configuration, the object captures domain-specific knowledge used to characterize a module.
With this information we know what settings were used in past analyses and can run the same analysis again in the future.
For example, we might reuse an analysis to compute a specific metric across a range of modules (to determine which module is best suited for our application) or to record the same analysis for a range of different module settings.

> On construction, a performance workflow requires an analysis configuration object that stores all settings required to run the analysis.

Reviewer: This line seems redundant with line 5

Author: I think this entire paragraph was left over from an earlier copy-paste. I've revised the intro text and push the changes.

   # (i.e. why the workflow is valid/invalid)
   print(validated_workflow.status_info)

Cross-validation can be performed on most predictors but will not produce results in the following scenarios:

- Training data contain simple mixtures from an :class:`~citrine.informatics.data_sources.AraTableDataSource` *and* an :class:`~citrine.informatics.predictors.IngredientsToSimpleMixturePredictor` is included in the graph.
- The graph includes a :class:`~citrine.informatics.predictors.SimpleMixturePredictor` or :class:`~citrine.informatics.predictors.GeneralizedMeanPropertyPredictor`.

> Both are known issues and will be resolved in a future release.

Reviewer: ```suggestion
Both are known issues in experimental functionality and will be resolved in a future release.
```

-  Support-weighted F1 score is calculated from averaged precision and recall of the model, weighted by the in-class fraction of true positives according to the formula ``2.0 * precision * recall / (precision + recall) * fraction_true_positives`` summed over each class.
   Scores are bounded by 0 and 1. At a value of 1, the model has perfect precision and recall.

In addition to the aforementioned metrics, predicted vs. actual data are also available.
The structure of the data will depend on whether the response is numeric or categorical.
For numeric responses, predicted and actual data contain the value and standard error associated with each data point.
For categorical responses, class probabilities are returned.

> (If desired, standard error of categorical responses can be computed using the equation ``1 / max(class_probability) - 1`` or other desired metric.)

Reviewer: ```suggestion
(If desired, an precision-based metric for categorical responses can be computed using the equation ``1 / max(class_probability) - 1``.)
```

Reviewer: I'm not sure what this line means.  If desired, the user can by themselves compute this?  Should we even be mentioning it (I don't think it stands on strong statistical footing)?  cc: @gregor-robinson

Author: Removed.
